addons/docs/src/blocks/Source.tsx

```diff
@@ -32,13 +58,45 @@ type NoneProps = CommonProps;
 
 type SourceProps = SingleSourceProps | MultiSourceProps | CodeProps | NoneProps;
 
+const getSnippet = (
+  storyId: StoryId,
+  sourceContext: SourceContextProps,
+  docsContext: DocsContextProps
+): string => {
+  const { sources } = sourceContext;
+  const { storyStore } = docsContext;
+
+  const snippet = sources && sources[storyId];
+  const data = storyStore?.fromId(storyId);
+
+  if (data) {
+    const { parameters } = data;
+    // eslint-disable-next-line no-underscore-dangle
+    const isArgsStory = parameters.__isArgsStory;
+    const type = parameters.docs?.source?.type || SourceType.AUTO;
+
+    // if user has explicitly set this as a dynamic story, or this is an args story and there's a snippet
+    if (
+      type === SourceType.DYNAMIC ||
+      (type === SourceType.AUTO && snippet && isArgsStory && !parameters.docs?.source?.code)
+    ) {
+      return snippet || '';
+    }
+
+    const enhanced = data && (enhanceSource(data) || data.parameters);
```

Comment:
Wouldn't parameters take preference?

Reply:
No, the enhancer gets the final say

Reply:
So why default/fallback to parameters then?

Reply:
The semantics of the enhancer is that it can return null if it wants to use the default. This is true for argType enhancers also

Reply:
Ok cool